cpp/src/arrow/config.h

```diff
@@ -77,4 +86,13 @@ const BuildInfo& GetBuildInfo();
 ARROW_EXPORT
 RuntimeInfo GetRuntimeInfo();
 
+struct ArrowGlobalOptions {
```

Comment:
Note sure it's worth putting "Arrow" here since this is in the `arrow` namespace.